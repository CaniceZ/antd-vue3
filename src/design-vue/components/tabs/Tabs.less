@border-default: 1px solid #f0f0f0;

.ygp-tabs {
  .ant-tabs-nav {
    .ant-tabs-tab[disabled] {
      color: #ebedf0;
      cursor: not-allowed;
    }

    margin: 0 0 8px 0;
  }

  &-fit {
    height: 100%;
    & > .ant-tabs-content-holder > .ant-tabs-content {
      height: 100%;
      & > .ant-tabs-tabpane {
        height: 100%;
      }
    }
  }

  &-border {
    & > .ant-tabs-nav {
      margin-bottom: 0;
    }
    & > .ant-tabs-content-holder {
      // padding-top: 8px;
      border-bottom: @border-default;
      border-left: @border-default;
      border-right: @border-default;
    }
  }

  & > .ant-tabs-content-holder > .ant-tabs-content {
    padding: var(--tabs-content-padding);
  }
}
