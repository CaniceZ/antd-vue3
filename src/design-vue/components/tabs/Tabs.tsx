import { Tabs as ATabs, TabsProps as ATabsProps } from 'ant-design-vue'
import { computed, defineComponent, PropType, ref, watch } from 'vue'
import { GlobalComponentConstructor, PropsType } from '../../types/ts-helpers'
import { getPrefix, suffixUnit } from '../../_utils/common'
import YTab from './Tab'
import { ATabsSlots } from './type'

function isBasicType(val) {
  return typeof val === 'string' || typeof val === 'number'
}

const [prefixName, prefixCls] = getPrefix('tabs')

const Tabs = defineComponent({
  name: prefixName,
  inheritAttrs: false,
  props: {
    modelValue: {
      type: [String, Number]
    },
    card: {
      type: Boolean,
      default: false
    },
    editableCard: {
      type: Boolean,
      default: false
    },
    options: {
      type: Array,
      default: () => []
    },
    propMap: {
      type: Object,
      default: () => ({
        label: 'name',
        value: 'type'
      })
    },
    disabled: {
      type: Boolean,
      default: false
    },
    gutter: Number as PropType<ATabsProps['tabBarGutter']>,
    fit: Boolean,
    bordered: Boolean,
    contentPadding: {
      type: [String, Number],
      default: 0
    }
  },
  emits: ['update:modelValue', 'change'],
  setup(props, { attrs, slots, emit }) {
    let initActiveKey = false
    const activeKey = ref(props.modelValue)
    watch(activeKey, val => {
      emit('update:modelValue', val)
      emit('change', val)
    })
    watch(
      () => props.modelValue,
      val => {
        activeKey.value = val
      }
    )
    const type = computed(() => {
      if (attrs.type) {
        return attrs.type as ATabsProps['type']
      } else if (props.card) {
        return 'card'
      } else if (props.editableCard) {
        return 'editable-card'
      } else {
        return 'line'
      }
    })
    const tabBarGutter = computed(() => {
      if (props.gutter !== void 0) {
        return props.gutter
      } else if (['card', 'editable-card'].includes(type.value!)) {
        return -1
      } else {
        return void 0
      }
    })
    return () => (
      <ATabs
        class={[
          prefixCls,
          {
            [`${prefixCls}-fit`]: props.fit,
            [`${prefixCls}-border`]: props.bordered
          }
        ]}
        style={`--tabs-content-padding: ${suffixUnit(props.contentPadding)};`}
        tabBarGutter={tabBarGutter.value}
        {...attrs}
        type={type.value}
        v-model:activeKey={activeKey.value}
        v-slots={{
          ...slots,
          default: () => {
            const childs = slots.default?.()
            if (childs) {
              const allKeyEmpty = childs.every(row => row.key === null)
              return childs.map((child, index) => {
                if (allKeyEmpty) {
                  child.key = index + 1
                }
                if (props.modelValue === void 0 && !initActiveKey) {
                  activeKey.value = child.key as any
                  initActiveKey = true
                }
                return child
              })
            } else {
              return props.options.map((row: any) => (
                <YTab
                  key={isBasicType(row) ? row : row[props.propMap.value]}
                  tab={isBasicType(row) ? row : row[props.propMap.label]}
                  disabled={
                    props.disabled
                      ? true
                      : typeof row === 'object' && row.disabled
                  }
                />
              ))
            }
          }
        }}
      />
    )
  }
})

export type TabsProps = PropsType<typeof Tabs> & PropsType<ATabsProps>
export type TabsExpose = {}
export type TabsSlots = ATabsSlots

export default Tabs as GlobalComponentConstructor<TabsProps, TabsSlots>
