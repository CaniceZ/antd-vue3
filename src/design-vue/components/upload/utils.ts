export const extname = (url = '') => {
  const temp = url.split('/')
  const filename = temp[temp.length - 1]
  const filenameWithoutSuffix = filename.split(/#|\?/)[0]
  return (/\.[^./\\]*$/.exec(filenameWithoutSuffix) || [''])[0]
}

const isImageFileType = type => !!type && type.indexOf('image/') === 0

export const isImageUrl = file => {
  if (isImageFileType(file.type)) {
    return true
  }
  const url = file.thumbUrl || file.url
  const extension = extname(url)
  if (
    /^data:image\//.test(url) ||
    /(webp|svg|png|gif|jpg|jpeg|jfif|bmp|dpg|ico)$/i.test(extension)
  ) {
    return true
  }
  if (/^data:/.test(url)) {
    // other file types of base64
    return false
  }
  if (extension) {
    // other file types which have extension
    return false
  }
  return true
}

export const getFileType = file => {
  if (!file) {
    return
  }
  const attachmentUrl = file.url

  const type = extname(attachmentUrl)
  let fileType = ''
  if (
    isImageFileType(file.type) ||
    /(webp|svg|png|gif|jpg|jpeg|jfif|bmp|dpg|ico)$/i.test(type)
  ) {
    fileType = 'image'
  } else if (type === '.pdf') {
    fileType = 'pdf'
  } else if (['.doc', '.dot'].includes(type)) {
    fileType = 'word'
  } else if (['.xls', '.xlsx'].includes(type)) {
    fileType = 'excel'
  } else {
    fileType = 'file'
  }
  return fileType
}

export const imageDimensions = file =>
  new Promise((resolve, reject) => {
    const img = new Image()

    img.onload = () => {
      const { naturalWidth: width, naturalHeight: height } = img
      resolve({ width, height })
    }

    img.onerror = () => {
      reject('There was some problem with the image.')
    }

    img.src = URL.createObjectURL(file)
  })

export const isUrl = (url: string) => {
  const regex = /^(http|https)/
  return regex.test(url.toLowerCase())
}
