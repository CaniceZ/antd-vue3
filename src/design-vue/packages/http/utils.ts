const toString = Object.prototype.toString

export function is(val: unknown, type: string) {
  return toString.call(val) === `[object ${type}]`
}

export function isString(val: unknown): val is string {
  return is(val, 'String')
}

export function isObject(val: any): val is Record<any, any> {
  return val !== null && is(val, 'Object')
}

export function isFunction(val: unknown): val is Function {
  return typeof val === 'function'
}

/**
 * Add the object as a parameter to the URL
 * @param baseUrl url
 * @param obj
 * @returns {string}
 * eg:
 *  let obj = {a: '3', b: '4'}
 *  setObjToUrlParams('www.baidu.com', obj)
 *  ==>www.baidu.com?a=3&b=4
 */
export function setObjToUrlParams(baseUrl: string, obj: any): string {
  let parameters = ''
  for (const key in obj) {
    parameters += key + '=' + encodeURIComponent(obj[key]) + '&'
  }
  parameters = parameters.replace(/&$/, '')
  return /\?$/.test(baseUrl)
    ? baseUrl + parameters
    : baseUrl.replace(/\/?$/, '?') + parameters
}

export function deepMerge<T = any>(src: any = {}, target: any = {}): T {
  let key: string
  for (key in target) {
    src[key] = isObject(src[key])
      ? deepMerge(src[key], target[key])
      : (src[key] = target[key])
  }
  return src
}

declare global {
  interface Navigator {
    msSaveBlob(blob: Blob, string): void
  }
}

/**
 * 根据后端接口文件流下载
 * @param {*} data
 * @param {*} filename
 * @param {*} mime
 * @param {*} bom
 */
export function downloadByData(
  data: BlobPart,
  filename: string,
  mime?: string,
  bom?: BlobPart
) {
  const blobData = typeof bom !== 'undefined' ? [bom, data] : [data]
  const blob = new Blob(blobData, { type: mime || 'application/octet-stream' })
  // 如果浏览器的msSaveBlob存在
  if (typeof window.navigator.msSaveBlob !== 'undefined') {
    // 使用msSaveBlob下载文件
    window.navigator.msSaveBlob(blob, filename)
  } else {
    // 创建一个URL
    const blobURL = window.URL.createObjectURL(blob)
    const tempLink = document.createElement('a')
    tempLink.style.display = 'none'
    tempLink.href = blobURL
    tempLink.setAttribute('download', filename)
    if (typeof tempLink.download === 'undefined') {
      tempLink.setAttribute('target', '_blank')
    }
    document.body.appendChild(tempLink)
    // 手动触发点击事件
    tempLink.click()
    // 点击完成后移除链接
    document.body.removeChild(tempLink)
    // 移出URL
    window.URL.revokeObjectURL(blobURL)
  }
}
