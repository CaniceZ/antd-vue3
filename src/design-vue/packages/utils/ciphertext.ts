import get from 'lodash-es/get'

type Options = {
  sign: '*' | string
  suffix: '_encrypt' | string
}

const defaultOptions: Options = {
  sign: '*',
  suffix: '_encrypt'
}

/**
 * 获取密文的密钥字段名
 * @param name
 * @param opts
 * @returns
 */
export function getCiphertextField(
  name: string,
  opts: Partial<Options> = defaultOptions
): string {
  const options = {
    ...defaultOptions,
    ...opts
  }
  return `${name}${options.suffix}`
}

/**
 * 获取密文对象实例
 * @param model 数据对象
 * @param name 密文字段名称 / 密文字段路径
 * @param opts.sign 脱敏显示符号
 * @param opts.suffix 脱敏密文字段名后缀
 * @returns
 */
export function getCiphertext(
  model: object,
  name: string,
  opts: Partial<Options> = defaultOptions
): {
  text: string
  cipherText: string
} {
  const options = {
    ...defaultOptions,
    ...opts
  }
  const emptyRes = {
    text: '',
    cipherText: ''
  }
  if (typeof model === 'object') {
    const text = get(model, name)
    const cipherText = get(model, getCiphertextField(name, options))
    if (text && cipherText && text.includes(options.sign)) {
      return {
        text,
        cipherText
      }
    } else {
      return emptyRes
    }
  } else {
    return emptyRes
  }
}

/**
 * 判断是否密文
 * @param model 数据对象
 * @param name 密文字段名称 / 密文字段路径
 * @param opts.sign 脱敏显示符号
 * @param opts.suffix 脱敏密文字段名后缀
 */
export function isCiphertext(
  model: object,
  name: string,
  opts: Partial<Options> = defaultOptions
): boolean {
  const options = {
    ...defaultOptions,
    ...opts
  }
  const ciphertext = getCiphertext(model, name, options)
  return Boolean(ciphertext.text && ciphertext.cipherText)
}
