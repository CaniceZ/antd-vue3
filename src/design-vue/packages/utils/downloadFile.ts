/**
 * 下载文件
 * @param url 下载路径
 * @returns void
 */
export default function downloadFile(url: string) {
  if (!url) return
  const a = document.createElement('a')
  a.href = url
  a.click()
}
