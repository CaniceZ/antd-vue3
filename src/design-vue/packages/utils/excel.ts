import { Message, Modal, YImportResultModal } from '../../'

/**
 * 导入文件
 * @param params
 * @param callback
 * @returns Promise
 */
export const importFile = async (params: {
  file: File | any
  title: string
  getBatchNumber: Function
  getImportResult: Function
  wrapCount?: number
  extraData?: Object
  templateCode?: string
  resultProps?: any
  resultSlots?: any
}) => {
  const {
    file,
    getBatchNumber,
    getImportResult,
    title,
    extraData,
    templateCode,
    resultProps,
    resultSlots,
    wrapCount
  } = params

  const formData = new FormData()
  formData.append('file', file)
  templateCode && formData.append('templateCode', templateCode) // 模板编码

  if (extraData) {
    Object.keys(extraData).forEach(key => {
      formData.append(key, extraData![key])
    })
  }

  const res: any = await getBatchNumber(formData)
  return new Promise((resolve, reject) => {
    const batchNumber = res?.batchNumber || res
    if (batchNumber) {
      Message.success('上传成功，稍后通知您导入结果!')
      importResultPolling({
        resultProps,
        resultSlots,
        getImportResult,
        batchNumber,
        title,
        wrapCount,
        doneCallback: () => {
          resolve(params)
        },
        timeoutCallback: () => {
          reject()
        }
      })
    } else {
      Message.error('上传失败，请稍后重试')
      reject()
    }
  })
}

/**
 * 导入结果轮询
 * @param params 轮询的请求参数
 * @param callback 成功回调
 */
export const importResultPolling = (params: {
  title: String
  batchNumber: string
  getImportResult: Function
  resultProps?: any
  resultSlots?: any
  wrapCount?: number
  doneCallback?: (...args: any) => void
  timeoutCallback?: (...args: any) => void
}) => {
  const {
    wrapCount,
    batchNumber,
    title,
    doneCallback,
    timeoutCallback,
    resultProps,
    resultSlots,
    getImportResult
  } = params
  if (!batchNumber) return
  function run(count = 1) {
    setTimeout(() => {
      let params = { batchNumber }
      if (resultProps?.extraData) {
        params = { ...params, ...resultProps.extraData }
      }
      const typeMap = resultProps?.typeMap
      getImportResult(params).then(
        res => {
          if (resultProps?.isPms) {
            // pms特殊导入结果处理
            if (res && res.id) {
              if (!resultProps?.hideModal) {
                Modal.create(YImportResultModal, {
                  list: [res],
                  typeMap,
                  importBizSet: [resultProps.importBiz],
                  title,
                  ...(resultProps || {})
                })
              }
              doneCallback?.({ res, typeMap, title })
            } else {
              if (count < (wrapCount || 20)) {
                run(++count)
              } else {
                Message.error('稍后请在最近导入结果中查看')
                timeoutCallback?.()
              }
            }
          } else {
            // excel通用
            const { status, msg } = res
            if (status === 4) {
              if (count < (wrapCount || 20)) {
                run(++count)
              } else {
                Message.error(msg || '稍后请在最近导入结果中查看')
                timeoutCallback?.()
              }
            } else {
              if (resultProps?.callback) {
                resultProps.callback(res)
              } else {
                if (!resultProps?.hideModal) {
                  Modal.create(
                    YImportResultModal,
                    {
                      title,
                      list: [res],
                      ...(resultProps || {})
                    },
                    resultSlots || {}
                  )
                }
                doneCallback?.({ res, typeMap, title })
              }
            }
          }
        },
        err => {
          Message.error(err?.msg || '上传失败，请稍后重试')
        }
      )
    }, 3000)
  }
  run()
}

// pms导出结果轮询
export const pmsExportPolling = (option: {
  batchNumber: string
  exportBiz: String
  userCode: String
  getPmsExportDetail: Function
  timeout?: (...args: any) => void
  done?: (...args: any) => void
  error?: (...args: any) => void
}) => {
  const {
    batchNumber,
    exportBiz,
    userCode,
    getPmsExportDetail,
    timeout,
    done,
    error
  } = option
  if (!batchNumber) return
  const params = {
    batchNumber: batchNumber,
    exportBiz,
    userCode
  }
  function run(count = 1) {
    setTimeout(() => {
      getPmsExportDetail(params).then(
        res => {
          const { code, data, msg } = res
          if (code === 0) {
            if (data && data.list.length > 0) {
              done?.(data)
              data.list &&
                data.list.forEach(item => {
                  if (item.excelUrl) {
                    window.location.href = item.excelUrl
                  } else {
                    Message.error(item.msg)
                  }
                })
            } else {
              if (count < 40) {
                run(++count)
              } else {
                Message.error('导出超时，请重新导出')
                timeout?.()
              }
            }
          } else {
            Message.error(msg)
            error?.()
          }
        },
        () => {
          error?.()
        }
      )
    }, 3000)
  }
  run()
}
