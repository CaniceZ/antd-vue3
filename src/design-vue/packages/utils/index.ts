import omit from './omit'
import pick from './pick'
import downloadFile from './downloadFile'
import { isCiphertext, getCiphertext, getCiphertextField } from './ciphertext'
import { needCLodop, loadCLodop, getLodop } from './LodopFuncs'
import { importFile, importResultPolling } from './excel'
import filterUnit from './filterUnit'
import filterMoney from './filterMoney'

export {
  omit,
  pick,
  downloadFile,
  isCiphertext,
  getCiphertext,
  getCiphertextField,
  needCLodop,
  loadCLodop,
  getLodop,
  importFile,
  importResultPolling,
  filterUnit,
  filterMoney
}
