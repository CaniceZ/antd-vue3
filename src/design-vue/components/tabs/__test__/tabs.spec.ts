import { mount } from '@vue/test-utils'
import { asyncExpect } from '../../../tests/utils'
import Tabs from '..'
import Ygpd from '../../../index'

describe('YTabs', () => {
  it('tabs component default slot render is ok', async () => {
    const wrapper = mount(Tabs as any, {
      global: {
        plugins: [Ygpd],
        renderStubDefaultSlot: true
      },
      slots: {
        default: `<y-tab-pane tab="选项一">选项一</y-tab-pane>
        <y-tab-pane tab="选项二">选项二</y-tab-pane>`
      },
      shallow: true
    })

    // 断言 默认插槽是否能够正常使用，如果能够获取我们传入的两个pane组件则证明传入正确，条件为真
    expect(wrapper.findAllComponents('y-tab-pane-stub').length === 2).toBe(true)

    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)

    // console.log(wrapper.html())
  })

  it('tabs component change is ok', async () => {
    const wrapper = mount(Tabs as any, {
      props: {
        card: true
      },
      global: {
        plugins: [Ygpd]
      },
      slots: {
        default: `<y-tab-pane key="1" tab="选项一">选项一</y-tab-pane>
        <y-tab-pane key="2" tab="选项二">选项二</y-tab-pane>`
      }
    })

    expect(wrapper.find('.ant-tabs-tab-active .ant-tabs-tab-btn').text()).toBe(
      '选项一'
    )

    await wrapper.setProps({ modelValue: '2' })

    expect(wrapper.find('.ant-tabs-tab-active .ant-tabs-tab-btn').text()).toBe(
      '选项二'
    )

    // 判断card：true有没有生效
    expect(wrapper.find('.ant-tabs-card').exists()).toBe(true)

    await asyncExpect(() => {
      expect(wrapper.html()).toMatchSnapshot()
    }, 0)
  })
})
