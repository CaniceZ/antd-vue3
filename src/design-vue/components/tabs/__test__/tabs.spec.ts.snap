// Vitest Snapshot v1

exports[`YTabs > tabs component change is ok 1`] = `
"<div class=\\"ant-tabs ant-tabs-top ant-tabs-card ygp-tabs\\">
  <div role=\\"tablist\\" class=\\"ant-tabs-nav\\">
    <!---->
    <div class=\\"ant-tabs-nav-wrap\\">
      <div class=\\"ant-tabs-nav-list\\" style=\\"transform: translate(0px, 0px);\\">
        <div class=\\"ant-tabs-tab\\">
          <div role=\\"tab\\" aria-selected=\\"false\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-1\\" aria-controls=\\"rc-tabs-test-panel-1\\">选项一</div>
          <!---->
        </div>
        <div class=\\"ant-tabs-tab ant-tabs-tab-active\\" style=\\"margin-left: -1px;\\">
          <div role=\\"tab\\" aria-selected=\\"true\\" class=\\"ant-tabs-tab-btn\\" tabindex=\\"0\\" id=\\"rc-tabs-test-tab-2\\" aria-controls=\\"rc-tabs-test-panel-2\\">选项二</div>
          <!---->
        </div>
        <!---->
        <div class=\\"ant-tabs-ink-bar ant-tabs-ink-bar-animated\\" style=\\"left: 0px; width: 0px;\\"></div>
      </div>
    </div>
    <div class=\\"ant-tabs-nav-operations ant-tabs-nav-operations-hidden\\">
      <!----><button type=\\"button\\" class=\\"ant-tabs-nav-more\\" style=\\"visibility: hidden; order: 1;\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" aria-haspopup=\\"listbox\\" aria-controls=\\"rc-tabs-test-more-popup\\" id=\\"rc-tabs-test-more\\" aria-expanded=\\"false\\"><span role=\\"img\\" aria-label=\\"ellipsis\\" class=\\"anticon anticon-ellipsis\\"><svg focusable=\\"false\\" class=\\"\\" data-icon=\\"ellipsis\\" width=\\"1em\\" height=\\"1em\\" fill=\\"currentColor\\" aria-hidden=\\"true\\" viewBox=\\"64 64 896 896\\"><path d=\\"M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z\\"></path></svg></span></button>
      <!---->
    </div>
    <!---->
    <!---->
  </div>
  <div class=\\"ant-tabs-content-holder\\">
    <div class=\\"ant-tabs-content ant-tabs-content-top\\">
      <div role=\\"tabpanel\\" tabindex=\\"-1\\" aria-hidden=\\"true\\" class=\\"ant-tabs-tabpane ygp-tab-pane\\" id=\\"rc-tabs-test-panel-1\\" aria-labelledby=\\"rc-tabs-test-tab-1\\" style=\\"display: none;\\">选项一</div>
      <div role=\\"tabpanel\\" tabindex=\\"0\\" aria-hidden=\\"false\\" style=\\"\\" class=\\"ant-tabs-tabpane ant-tabs-tabpane-active ygp-tab-pane\\" id=\\"rc-tabs-test-panel-2\\" aria-labelledby=\\"rc-tabs-test-tab-2\\">选项二</div>
    </div>
  </div>
</div>"
`;

exports[`YTabs > tabs component default slot render is ok 1`] = `
"<a-tabs-stub type=\\"line\\" animated=\\"[object Object]\\" tabposition=\\"top\\" destroyinactivetabpane=\\"false\\" hideadd=\\"false\\" centered=\\"false\\" class=\\"ygp-tabs\\" activekey=\\"1\\">
  <y-tab-pane-stub tab=\\"选项一\\">选项一</y-tab-pane-stub>
  <y-tab-pane-stub tab=\\"选项二\\">选项二</y-tab-pane-stub>
</a-tabs-stub>"
`;
